src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -740,6 +743,44 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     })
   }
 
+  override def listUserResourcesWithRolesAndActions(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Iterable[ResourceIdWithRolesAndActions]] = {
```

Comment:
@adrazhi this PR rewrites the queries sam uses to determine which terra resource a user may access. I marked the associated JIRA issue as high security impact for this reason. The large additions to this file are the new queries.